cluster/kubernetes/kubernetes.go

```diff
@@ -91,21 +103,9 @@ type Cluster struct {
 }
 
 // NewCluster returns a usable cluster.
-func NewCluster(clientset k8sclient.Interface,
-	dynamicClientset k8sclientdynamic.Interface,
-	fluxHelmClientset fhrclient.Interface,
-	applier Applier,
-	sshKeyRing ssh.KeyRing,
-	logger log.Logger,
-	nsWhitelist []string,
-	imageExcludeList []string) *Cluster {
-
+func NewCluster(client ExtendedClient, applier Applier, sshKeyRing ssh.KeyRing, logger log.Logger, nsWhitelist []string, imageExcludeList []string) *Cluster {
```

Comment:
Nice!

Reply:
It's a lot tidier, thanks for the suggestion. We could also

 - make NewCluster accept an interface with only the client API methods it uses (though this would mean gathering all the various bits of clientset, verbatim; fiddly, but maybe worth it for being precise about the requirements of the cluster implementation)
 - use an options value for the assorted other arguments passed to NewCluster (`nsWhitelist` and `imageExcludeList` e.g.)